### PR TITLE
Implemented Page Eviction, Join and StringAggregator

### PR DESCRIPTION
**Pull request by Priiyam**
#21 #19 #18 
I and Ram collaborated with each other to first finish off Exercise 1 and 2 which took quite a lot of phone calls to :P Then Manish and Ritesh helped with getting the insertTuple() and deleteTuple() in BufferPool along with all the other dependencies including writePage() required for Page Eviction.

### CODE REVIEW

**Review by Ram-Aditya — APPROVED** [review]

**Review by Rithesh17 — APPROVED** [review]

**Comment by sbshah97** [comment]
Why has this Pull Request not been merged?

**Review by manishsbhoopalam — APPROVED** [review]

**Comment by Ram-Aditya** [comment]
@salman-bhai We thought we needed to wait for your approval as we had requested a review from your side.
Sorry about that!We `ll go ahead with the merge now then.

**Pull request merged**
